src/main/java/me/anxuiz/settings/bukkit/plugin/gui/clickables/Clickable.java

```diff
@@ -0,0 +1,28 @@
+package me.anxuiz.settings.bukkit.plugin.gui.clickables;
+
+import me.anxuiz.settings.bukkit.plugin.BukkitSettingsPlugin;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+
+public abstract class Clickable implements Click, Listener {
```

Comment:
I didn't even think of the impact of having a listener for each border icon, which I can see how efficient that would be. I'll move the handling of border interaction into a single event handler. It was also a mistake that I didn't check what the inventory was before unregistering. However I'm not really sure how else to safeguard against the listener being registered forever? A delayed task could be used to automatically unregister the listener after a short period of time?